dm/dm.go

```diff
@@ -72,13 +73,15 @@ var usage = func() {
 	fmt.Fprintln(os.Stderr, "Flags:")
 	flag.PrintDefaults()
 	fmt.Fprintln(os.Stderr)
-	os.Exit(1)
+	fmt.Fprintln(os.Stderr, "--stdin requires a file name and either the file contents or a tar archive containing the named file.")
+	fmt.Fprintln(os.Stderr, "        a tar archive may include any additional files referenced directly or indirectly by the named file.")
+	panic("\n")
 }
 
 func getGitRegistry() *registry.GithubRegistry {
 	s := strings.Split(*template_registry, "/")
 	if len(s) < 2 {
-		log.Fatalf("invalid template registry: %s", *template_registry)
+		panic(fmt.Errorf("invalid template registry: %s", *template_registry))
```

Comment:
Maybe I'm the one being obtuse here. I like using panic, because I see it used widely in the golang libraries and tools as an idiomatic way of dealing with serious failures.

In this case, I like using it because it allows the top level of tool to maintain control over the shut down. Granted, we aren't doing anything there except logging the error and exiting with a non-zero error code. And, granted, we could do the same at each of the failure sites using log.Fatalf.

However, using log.Fatalf or os.Exit anywhere other than the top level means that higher level stack frames have no opportunity to clean things up. Currently, the tool is simple enough that we can probably just die locally. However, when adding the first cut of reading from stdin, I needed to remove a temporary directory, and found that the tool exited locally all over the place.

It just feels to me like best practice and the established idiom to unwind the stack in Go, not to mention the widespread use of exceptions in languages like Java and C# where you'd never dream of just killing the process way down the stack.

The only difference, IMO, between panic and exception is the name, which sound more dire in the case of panic. However, the mechanisms are nearly identical. Both unwind the stack in a way that lets callers clean up. Both can be handled easily. Both print a stack trace if not handled at the top level.

That said, if you feel strongly that it's just not right, then I'll take it out. This is community code, after all, and as stakeholders, the stylistic approach adopted by the code should make all of us comfortable.